pkg/controller/logstorage/kubecontrollers/es_kube_controllers.go

```diff
@@ -363,6 +352,15 @@ func (r *ESKubeControllersController) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// Get secrets needed for kube-controllers to talk to elastic. This is needed for zero-tenants and single-tenants
+	// that deploy es-kube-controllers and need to talk to es-gateway
+	var kubeControllersUserSecret *core.Secret
```

Comment:
kubeControllersUserSecret is created in tigera-operator namespace only in createESGateway func in line 327, Hence moving the fetch logic after that.